tests/decode_requests.rs

```diff
@@ -38,3 +37,47 @@ fn decode_request_api_version() -> Result<(), DecodeError> {
 
     Ok(())
 }
+
+#[test]
+fn decode_request_metadata() -> Result<(), DecodeError> {
+    const KAFKA_REQUEST_METADATA: &[u8] = &[
+        // Request/response Size => INT32
+        0, 0, 0, 24, // value: 24
+        // api_key => INT16
+        0, 3, // value: 3
+        // api_version => INT16
+        0, 2, // value: 2
+        // correlation_id => INT32
+        0, 0, 0, 1, // value: 1
+        // client_id => NULLABLE_STRING
+        0, 10, 99, 111, 110, 115, 117, 109, 101, 114, 45, 49, // len: 10, value: consumer-1
+        // topics => STRING
+        0, 2, 72, 105, // value: "Hi"
+    ];
+
+    use std::io::Cursor;
+    let mut buf = Cursor::new(KAFKA_REQUEST_METADATA);
+    let request = DecodedRequest::decode(&mut buf)?;
+
+    assert_eq!(24, request.size);
+    assert_eq!(
+        HeaderRequest {
+            api_key: 3,
+            api_version: 2,
+            correlation_id: 1,
+            client_id: String::from("consumer-1")
+        },
+        request.header
+    );
+    assert_eq!(
+        BodyRequest::Metadata(BodyMetadataRequest {
+            // ToDo: If we make 'topics' field private, then it cannot be tested here,
```

Comment:
If the private fields have accessor methods, I'd probably just test each expected value individually, as in:
```rust
assert_eq!(body.topics(), "hi");
assert_eq!(body.is_auto_topic_creation_allowed(), false);
```
This isn't as pretty, but it doesn't require fields to be public. Also, IMO, you get better failure messages --- it's clearer which field had an unexpected value.